airlock/views/request.py

```diff
@@ -650,40 +651,53 @@ def multiselect_withdraw_files(request, multiform, release_request):
 
 
 def multiselect_update_files(request, multiform, release_request):
-    files_to_add = []
+    files_to_add = {}
     files_ignored = {}
+    filegroup = None
     # validate which files can be added
     for f in multiform.cleaned_data["selected"]:
         workspace = bll.get_workspace(release_request.workspace, request.user)
-        relpath = release_request.get_request_file_from_urlpath(f).relpath
-        if permissions.user_can_change_request_file_group(
-            request.user, release_request, workspace, relpath
+        request_file = release_request.get_request_file_from_urlpath(f)
+        if filegroup is None:
+            filegroup = request_file.group
+        else:
+            # We should always be updating files from the same group
+            assert filegroup == request_file.group
```

Comment:
Would it be cleaner to have `filegroup = multiform.cleaned_data["selected"][0].group` outside the loop?  You then don't need to wrap the assertion in an `if` branch (which is something I'm surprised coverage hasn't moaned about).

Reply:
I didn't do this before, because you can't do `multiform.cleaned_data["selected"][0].group` directly, so it would mean repeating the `request_file = release_request.get_request_file_from_urlpath()` call.

It could instead set the filegroup on the first iteration and the assert the current iteration's request file's group matches. Is that better?
```
    for i, f in enumerate(multiform.cleaned_data["selected"]):
        workspace = bll.get_workspace(release_request.workspace, request.user)
        request_file = release_request.get_request_file_from_urlpath(f)
        if i == 0:
            filegroup = request_file.group
        # We should always be updating files from the same group
        assert request_file.group == filegroup
```

Reply:
Fine.